neqo-transport/src/cc/classic_cc.rs

```diff
@@ -562,153 +514,186 @@ mod tests {
         )
     }
 
-    fn persistent_congestion(lost_packets: &[SentPacket]) -> bool {
-        let mut cc = ClassicCongestionControl::new(NewReno::default());
-        for p in lost_packets {
-            cc.on_packet_sent(p);
+    fn congestion_control(cc: CongestionControlAlgorithm) -> Box<dyn CongestionControl> {
+        match cc {
+            CongestionControlAlgorithm::NewReno => {
+                Box::new(ClassicCongestionControl::new(NewReno::default()))
+            }
+            CongestionControlAlgorithm::Cubic => {
+                Box::new(ClassicCongestionControl::new(Cubic::default()))
+            }
         }
+    }
 
-        cc.on_packets_lost(Some(now()), None, PTO, lost_packets);
-        if cc.cwnd() == CWND_INITIAL / 2 {
-            false
-        } else if cc.cwnd() == CWND_MIN {
-            true
-        } else {
-            panic!("unexpected cwnd");
+    fn persistent_congestion(lost_packets: &[SentPacket], persistent_expected: bool) {
+        for cc_alg in &[
+            CongestionControlAlgorithm::NewReno,
+            CongestionControlAlgorithm::Cubic,
+        ] {
```

Comment:
I will leave as it is. I implemented something similar before, but some of the test functions that use the function got a bit hard to read because the function is called with a different  cc algorithm and multiple different configurations.

Reply:
I'm not sure if we aren't talking past each other, so I will propose a change.